src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -255,6 +259,7 @@ void GcsNodeManager::OnNodeFailure(const NodeID &node_id) {
   if (auto node = RemoveNode(node_id, /* is_intended = */ false)) {
     node->set_state(rpc::GcsNodeInfo::DEAD);
     node->set_end_time_ms(current_sys_time_ms());
+    node->mutable_death_info()->set_reason(rpc::NodeDeathInfo::UNEXPECTED_TERMINATION);
```

Comment:
we should pass the argument here?

Reply:
The detail has the autoscaler.DrainNodeReason though